crates/blockchain/blockchain.rs

```diff
@@ -161,6 +148,109 @@ pub fn validate_block(
     Ok(())
 }
 
+pub fn is_canonical(
+    store: &Store,
+    block_number: BlockNumber,
+    block_hash: BlockHash,
+) -> Result<bool, StoreError> {
+    match store.get_canonical_block_hash(block_number)? {
+        Some(hash) if hash == block_hash => Ok(true),
+        _ => Ok(false),
+    }
+}
+
+pub fn new_head(
```

Comment:
it's not clear what this function those from the name, can you add a comment?

Reply:
Added comments and changed name to `apply_fork_choice`